sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseBodyDecoder.java

```diff
@@ -274,90 +234,46 @@ private static Object deserializePage(String value, Type resultType, Type wireTy
+            } else if (wireType == UnixTime.class) {
+                result = ((UnixTime) wireResponse).getDateTime();
+            }
+        } else if (TypeUtil.isTypeOrSubTypeOf(resultType, List.class)) {
+            final Type resultElementType = TypeUtil.getTypeArgument(resultType);
+
+            @SuppressWarnings("unchecked") final List<Object> wireResponseList = (List<Object>) wireResponse;
+
+            final int wireResponseListSize = wireResponseList.size();
+            for (int i = 0; i < wireResponseListSize; ++i) {
+                final Object wireResponseElement = wireResponseList.get(i);
+                final Object resultElement =
+                    convertToResultType(wireResponseElement, resultElementType, wireType);
+                if (wireResponseElement != resultElement) {
+                    wireResponseList.set(i, resultElement);
                 }
-            } else {
-                if (TypeUtil.isTypeOrSubTypeOf(resultType, List.class)) {
-                    final Type resultElementType = TypeUtil.getTypeArgument(resultType);
-
-                    @SuppressWarnings("unchecked") final List<Object> wireResponseList = (List<Object>) wireResponse;
-
-                    final int wireResponseListSize = wireResponseList.size();
-                    for (int i = 0; i < wireResponseListSize; ++i) {
-                        final Object wireResponseElement = wireResponseList.get(i);
-                        final Object resultElement =
-                            convertToResultType(wireResponseElement, resultElementType, wireType);
-                        if (wireResponseElement != resultElement) {
-                            wireResponseList.set(i, resultElement);
-                        }
-                    }
-                    //
-                    result = wireResponseList;
-                } else if (TypeUtil.isTypeOrSubTypeOf(resultType, Map.class)) {
-                    final Type resultValueType = TypeUtil.getTypeArguments(resultType)[1];
-
-                    @SuppressWarnings("unchecked") final Map<String, Object> wireResponseMap =
-                        (Map<String, Object>) wireResponse;
-
-                    final Set<Map.Entry<String, Object>> wireResponseEntries = wireResponseMap.entrySet();
-                    for (Map.Entry<String, Object> wireResponseEntry : wireResponseEntries) {
-                        final Object wireResponseValue = wireResponseEntry.getValue();
-                        final Object resultValue = convertToResultType(wireResponseValue, resultValueType, wireType);
-                        if (wireResponseValue != resultValue) {
-                            wireResponseMap.put(wireResponseEntry.getKey(), resultValue);
-                        }
-                    }
-                    //
-                    result = wireResponseMap;
-                } else if (TypeUtil.isTypeOrSubTypeOf(resultType, PagedResponseBase.class)) {
```

Comment:
Removed as this should be caught by the check for `Page.class` and should be deserialized through that code path.